### ISSUE TITLE
AArch64 macOS JDK 11: Failures in SharedClasses.SCM23.MultiThread* tests

### ISSUE THREAD

**Issue by knn-k** [eclipse-openj9/openj9#14621]
Failure link
------------
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_extended.system_aarch64_mac_Nightly_testList_0/1/
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_extended.system_aarch64_mac_Nightly_testList_1/1/

- SharedClasses.SCM23.MultiThread_0
- SharedClasses.SCM23.MultiThreadMultiCL_0

Failure output (captured from console output)
---------------------------------------------

```
MTM2 18:49:55 >> Loaded 20000 classes...
MTM2 stderr Unhandled exception
MTM2 stderr Type=Segmentation error vmState=0x00000000
MTM2 stderr J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000002
MTM2 stderr Handler1=000000010304FF14 Handler2=0000000103322E38 InaccessibleAddress=000000012AFFFFF8
MTM2 stderr x0=000000012B009D00 x1=000000012B009D00 x2=000000037FC0C728 x3=000000037FC0C740
MTM2 stderr x4=000000037FC0C740 x5=0000000000000000 x6=000000010857E338 x7=0000000000000000
MTM2 stderr x8=00000001151978BD x9=000000012B000020 x10=000000012B000020 x11=0000000000000000
MTM2 stderr x12=000000016D156260 x13=000000016D156280 x14=000000037FC0C784 x15=000000016D156600
MTM2 stderr x16=000000010309418C x17=00000002067AD648 x18=0000000000000000 x19=000000012B009D00
MTM2 stderr x20=000000012B000020 x21=000000016D156538 x22=000000012B00BC68 x23=000000014D80A820
MTM2 stderr x24=000000016D156480 x25=00000001084D0804 x26=000000016D156480 x27=0000000108455160
MTM2 stderr x28=000000014D80A820 x29(FP)=000000016D156290 x30(LR)=0000000107D99064 x31(SP)=000000016D156280
MTM2 stderr PC=0000000107D84C70 SP=000000016D156280
MTM2 stderr v0 0000000000000005 (f: 5.000000, d: 2.470328e-323)
MTM2 stderr v1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v2 0706050403020100 (f: 50462976.000000, d: 7.949929e-275)
MTM2 stderr v3 bfc95a5a5cf7013f (f: 1559691520.000000, d: -1.980699e-01)
MTM2 stderr v4 0000000000000003 (f: 3.000000, d: 1.482197e-323)
MTM2 stderr v5 0000000000000002 (f: 2.000000, d: 9.881313e-324)
MTM2 stderr v6 0000080000000800 (f: 2048.000000, d: 4.345847e-311)
MTM2 stderr v7 000000000000000e (f: 14.000000, d: 6.916919e-323)
MTM2 stderr v8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v16 bfd0000000000000 (f: 0.000000, d: -2.500000e-01)
MTM2 stderr v17 3fd566b47b355555 (f: 2067092864.000000, d: 3.343936e-01)
MTM2 stderr v18 bf71689aec27fcc3 (f: 3962043648.000000, d: -4.250150e-03)
MTM2 stderr v19 3fe62e42fefa39ef (f: 4277811712.000000, d: 6.931472e-01)
MTM2 stderr v20 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v21 ffffffffffffffff (f: 4294967296.000000, d: nan)
MTM2 stderr v22 ffffffffffffffff (f: 4294967296.000000, d: nan)
MTM2 stderr v23 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v24 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v25 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v26 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v27 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v28 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v29 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v30 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr v31 0000000000000000 (f: 0.000000, d: 0.000000e+00)
MTM2 stderr Module=/Users/jenkins/workspace/Test_openjdk11_j9_extended.system_aarch64_mac_Nightly_testList_0/openjdkbinary/j2sdk-image/lib/default/libj9jit29.dylib
MTM2 stderr Module_base_address=0000000107800000 Symbol=old_slow_jitHandleInternalErrorTrap
MTM2 stderr Symbol_address=0000000107D84C54
MTM2 stderr Target=2_90_20220301_1 (Mac OS X 11.4)
MTM2 stderr CPU=aarch64 (8 logical CPUs) (0x400000000 RAM)
```

**Comment by knn-k** [comment]
I opened https://github.com/adoptium/aqa-tests/pull/3391 for disabling SharedClasses.SCM23.MultiThread* tests on AArch64 macOS for the time being.

**Comment by knn-k** [comment]
These tests did not fail with JDK 18 v.0.31.0 branch last week: https://openj9-jenkins.osuosl.org/job/Test_openjdk18_j9_extended.system_aarch64_mac_Release/1/

**Comment by knn-k** [comment]
This happened also in JDK next: https://openj9-jenkins.osuosl.org/job/Test_openjdknext_j9_extended.system_aarch64_mac_Nightly/1/

**Comment by knn-k** [comment]
Loading the core file from the job above indo lldb generates the following warnings, and I cannot get much information from it:
```
warning: (arm64e) /Users/openj9/20220307-173258-SharedClasses/results/core.20220307.173401.83227.0001.dmp load command 486 LC_SEGMENT_64 has a fileoff + filesize (0x23b7623d0) that extends beyond the end of the file (0x1fb7623d0), the segment will be truncated to match
warning: (arm64e) /Users/openj9/20220307-173258-SharedClasses/results/core.20220307.173401.83227.0001.dmp load command 487 LC_SEGMENT_64 has a fileoff (0x23b7623d0) that extends beyond the end of the file (0x1fb7623d0), ignoring this section
(lldb) thread list
Process 0 stopped
* thread #1: tid = 0x0000, stop reason = signal SIGSTOP
  thread #2: tid = 0x0001, stop reason = signal SIGSTOP
  thread #3: tid = 0x0002, stop reason = signal SIGSTOP
  thread #4: tid = 0x0003, stop reason = signal SIGSTOP
  thread #5: tid = 0x0004, stop reason = signal SIGSTOP
  thread #6: tid = 0x0005, stop reason = signal SIGSTOP
  thread #7: tid = 0x0006, stop reason = signal SIGSTOP
  thread #8: tid = 0x0007, stop reason = signal SIGSTOP
  thread #9: tid = 0x0008, stop reason = signal SIGSTOP
(lldb) bt
* thread #1, stop reason = signal SIGSTOP
  frame #0: 0xffffffffffffffff
```

**Comment by knn-k** [comment]
PR #15907 should fix this issue.

**Comment by knn-k** [comment]
No failures in Grinder jobs with PR #15907: https://openj9-jenkins.osuosl.org/job/Grinder/1274/ and https://openj9-jenkins.osuosl.org/job/Grinder/1275/

**Comment by pshipton** [comment]
@knn-k https://github.com/adoptium/aqa-tests/pull/3969 is still in draft state, are you going to make it ready for review?

**Comment by knn-k** [comment]
@pshipton I changed the status of https://github.com/adoptium/aqa-tests/pull/3969.  I was thinking of making it ready for review after #15924 is merged into the v0.35.0 branch.

**Comment by pshipton** [comment]
0.35 testing won't be affected until the weekend. We want to see the results of merging https://github.com/eclipse-openj9/openj9/pull/15907 in the head stream before merging https://github.com/eclipse-openj9/openj9/pull/15924.

**Comment by knn-k** [comment]
The tests were enabled, and they ran successfully in the latest nightly job: https://openj9-jenkins.osuosl.org/job/Test_openjdk17_j9_extended.system_aarch64_mac_Nightly/145/

Closing.

**Comment by pshipton** [comment]
@knn-k when tests are re-enabled, pls remove the `test excluded` label. We don't want to find any closed issues that still have this label.
